yellowbrick/contrib/missing/base.py

```diff
@@ -0,0 +1,69 @@
+# yellowbrick.contrib.missing.base
+# Base Visualizer for missing values
+#
+# Author:  Nathan Danielsen <nathan.danielsen@gmail.com>
+# Created: Fri Mar 29 5:17:36 2018 -0500
+#
+# Copyright (C) 2018 District Data Labs
+# For license information, see LICENSE.txt
+#
+# ID: base.py [] nathan.danielsen@gmail.com.com $
+
+"""
+Base classes for missing values visualizers.
+"""
+
+##########################################################################
+## Imports
+##########################################################################
+import numpy as np
+
+from yellowbrick.features.base import DataVisualizer
+from yellowbrick.utils import is_dataframe
+
+##########################################################################
+## Feature Visualizers
+##########################################################################
+
+class MissingDataVisualizer(DataVisualizer):
+    """Base class for MissingDataVisualizers.
+    """
+
+    def fit(self, X, y=None, **kwargs):
+        """
+        The fit method is the primary drawing input for the
+        visualization since it has both the X and y data required for the
+        viz and the transform method does not.
+
+        Parameters
+        ----------
+        X : ndarray or DataFrame of shape n x m
+            A matrix of n instances with m features
+
+        y : ndarray or Series of length n
+            An array or series of target or class values
+
+        kwargs : dict
+            Pass generic arguments to the drawing method
+
+        Returns
+        -------
+        self : instance
+            Returns the instance of the transformer/visualizer
+        """
+        if is_dataframe(X):
+            self.X = X.values
+            if self.features_ is None:
+                self.features_ = X.columns
+        else:
+            self.X = X
+
+        self.y = y
```

Comment:
I'm a bit squeamish storing X and y on the visualizer - for a lot of workflows they could be fairly large arrays that aren't intended to be held in memory for longer than it takes to transform them (e.g. multiple visualizers would duplicate the arrays in memory), and they refer to data that's being passed through transformers and manipulated, so if stored by reference and not copied, these values would change as the data moved to the next stage of processing. 

It would be my preference that in the bar and dispersion plots, they override `fit()` where most of the work is conducted, including calls to `get_nan_values` that pass the data in as arguments. It would make the class a bit easier to test as well.